File: src/main/java/ninekothecat/catconomy/defaultImplementations/CatTransaction.java
package ninekothecat.catconomy.defaultImplementations;

import ninekothecat.catconomy.enums.TransactionType;
import ninekothecat.catconomy.interfaces.ITransaction;
import org.bukkit.plugin.Plugin;

import java.util.ArrayList;
import java.util.Collection;
import java.util.UUID;

public class CatTransaction implements ITransaction {
    private final TransactionType transactionType;
    private final boolean isConsole;
    private final double amount;
    private final UUID initiator;
    private final ArrayList<UUID> usersInvolved;
    private final Plugin plugin;
    private final String message;

    public CatTransaction(TransactionType transactionType, boolean isConsole, double amount, UUID initiator, ArrayList<UUID> usersInvolved,  String message,Plugin plugin) {
        this.transactionType = transactionType;
        this.isConsole = isConsole;
        this.amount = amount;
        this.initiator = initiator;
        this.usersInvolved = usersInvolved;
        this.plugin = plugin;
        this.message = message;
    }

    @Override
    public Plugin getPlugin() {
        return plugin;
    }

    @Override
    public String getMessage() {
        return message;
    }

    @Override
    public TransactionType getTransactionType() {
        return this.transactionType;
    }

    @Override
    public boolean isConsole() {
        return this.isConsole;
    }

    @Override
    public UUID getInitiator() {
        return this.initiator;
    }

    @Override
    public Collection<UUID> getUsersInvolved() {
        return this.usersInvolved;
    }

    @Override
    public Double getAmount() {
        return this.amount;
    }
}

File: src/main/kotlin/ninekothecat/catconomy/defaultImplementations/CatTransaction.kt
package ninekothecat.catconomy.defaultImplementations

import ninekothecat.catplugincore.money.enums.TransactionType
import ninekothecat.catplugincore.money.interfaces.ITransaction
import org.bukkit.plugin.Plugin
import java.util.*

class CatTransaction(
    override val transactionType: TransactionType,
    override val isConsole: Boolean,
    override val amount: Double,
    override val initiator: UUID?,
    override val usersInvolved: ArrayList<UUID>,
    override val message: String,
    override val plugin: Plugin
) : ITransaction